test/parallel/test-runner-run-coverage.mjs

```diff
@@ -0,0 +1,207 @@
+import * as common from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.mjs';
+import { describe, it, run } from 'node:test';
+import assert from 'node:assert';
+
+const files = [fixtures.path('test-runner', 'coverage.js')];
+
+describe('require(\'node:test\').run Coverage settings', { concurrency: true }, async () => {
+  await describe('validation', async () => {
+    await it('should only allow boolean in options.coverage', async () => {
+      [Symbol(), {}, () => {}, 0, 1, 0n, 1n, '', '1', Promise.resolve(true), []]
+        .forEach((coverage) => assert.throws(() => run({ coverage }), {
+          code: 'ERR_INVALID_ARG_TYPE'
+        }));
+    });
+
+    await it('should only allow coverage options when coverage is true', async () => {
+      assert.throws(
+        () => run({ coverage: false, coverageIncludeGlobs: [] }),
+        { code: 'ERR_INVALID_ARG_VALUE' },
+      );
+      assert.throws(
+        () => run({ coverage: false, coverageExcludeGlobs: [] }),
+        { code: 'ERR_INVALID_ARG_VALUE' },
+      );
+      assert.throws(
+        () => run({ coverage: false, lineCoverage: 0 }),
+        { code: 'ERR_INVALID_ARG_VALUE' },
+      );
+      assert.throws(
+        () => run({ coverage: false, branchCoverage: 0 }),
+        { code: 'ERR_INVALID_ARG_VALUE' },
+      );
+      assert.throws(
+        () => run({ coverage: false, functionCoverage: 0 }),
+        { code: 'ERR_INVALID_ARG_VALUE' },
+      );
+    });
+
+    await it('should only allow string|string[] in options.coverageExcludeGlobs', async () => {
+      [Symbol(), {}, () => {}, 0, 1, 0n, 1n, Promise.resolve([]), true, false]
+        .forEach((coverageExcludeGlobs) => {
+          assert.throws(() => run({ coverage: true, coverageExcludeGlobs }), {
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+          assert.throws(() => run({ coverage: true, coverageExcludeGlobs: [coverageExcludeGlobs] }), {
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+        });
+      run({ files: [], signal: AbortSignal.abort(), coverage: true, coverageExcludeGlobs: [''] });
+      run({ files: [], signal: AbortSignal.abort(), coverage: true, coverageExcludeGlobs: '' });
```

Comment:
To save on allocations, can we create the `AbortSignal.abort()` only once?

Reply:
This is a UT, do we actually care about allocating another AbortSignal?

I prefer keeping it simple 🤷🏽‍♂️

Reply:
Our CI runs thousands of tests and we have existing memory issues with some of the runners. Every bit counts. It's not critical but would be nice. Ok to ignore tho.